### PR TITLE
Update DiscussionListItem.less to fix touch double tap

### PR DESCRIPTION
**Pull request by zerosonesfun**
Adds rule for devices without hover capabilities which hides discussion item toggle button. This prevents having to tap twice to open the discussion. Also, the toggle button is not used on mobile devices and so it should be completely hidden. Additional nested rules could be added here in the future if needed. 

**Fixes https://github.com/flarum/core/issues/1913**

**Changes proposed in this pull request:**
Adds rule for devices without hover capabilities which hides discussion item toggle button. This prevents having to tap twice to open the discussion. Also, the toggle button is not used on mobile devices and so it should be completely hidden. Additional nested rules could be added here in the future if needed. See issue: https://github.com/flarum/core/issues/1913

Fix uses a relatively new hover CSS media feature which can be used to test whether the user’s primary input mechanism can hover over elements. It is compatible will all major/modern browsers. No Internet Explorer support but on August 17th, 2021, Internet Explorer 11 will no longer be supported by Microsoft. All IE versions are on their way out. Also, if someone visits a forum with this one browser, the worst that would happen is they would have to double tap like before because the browser wouldn’t recognize the rule. 

**Reviewers should focus on:**
If reviewer does not have an iPhone or iPad, reviewer could at least confirm that this fix does not affect other devices. Toggle button should still work on desktop. If reviewer has an iPhone or iPad they will be able to open a discussion with one tap. 

**Confirmed**

- [x] Frontend changes: tested on a local Flarum installation.
- [x] Backend changes: tests are green (run `composer test`).

### CODE REVIEW

**Comment by askvortsov1** [comment]
When I tested this locally, it removed it on desktop as well for me. Same with pointer: none. Not sure why though?

**Comment by zerosonesfun** [comment]
> When I tested this locally, it removed it on desktop as well for me. Same with pointer: none. Not sure why though?

Does your Desktop have any touch or touch-like features? It looks like I should have used “any-hover” instead of “hover.” Any hover would say, “if this device has any normal hover abilities at all, don’t hide the toggle button. 

https://developer.mozilla.org/en-US/docs/Web/CSS/@media/any-hover

**Comment by askvortsov1** [comment]
Maybe it's just my computer being weird? Either way, it still considers my computer an `any-hover: none` device. Very odd...

**Comment by zerosonesfun** [comment]
> Maybe it's just my computer being weird? Either way, it still considers my computer an `any-hover: none` device. Very odd...

Weird. But you can see that the any-hover rule is being applied? Your computer wishes it was a touch screen. It’s got touch screen envy. 

I was trying to avoid this fix relying on screen width. Because there’s so many different sizes of tablets.

**Comment by zerosonesfun** [comment]
There are other options to try as well:

https://css-tricks.com/touch-devices-not-judged-size/

**Review by askvortsov1 — APPROVED** [review]
I have confirmed that the issues I experienced are due to an issue with my Wayland setup (and even then, only occured on chrome). I'm in favor of merging this.

@zerosonesfun The issue also mentioned some problems with notifications; would you be able to include a fix for that here as well?

**Comment by zerosonesfun** [comment]
> I have confirmed that the issues I experienced are due to an issue with my Wayland setup (and even then, only occured on chrome). I'm in favor of merging this.
> 
> @zerosonesfun The issue also mentioned some problems with notifications; would you be able to include a fix for that here as well?

I started to add a change for the notifications but then I removed the change and put the file back to the way it was because I realized that notifications probably should stay as-is for now. The reason is, the first time makes a little mark as read check mark appear. Then you can tap the check mark to mark the notification as read without having to visit the discussion if you don’t want to. So, although a little unintuitive, I don’t think the notification double tap on mobile is an “issue” per se. It’s kind of a feature. You can tap once, then you have the option of marking as read, or going to the discussion. 

What should be done down the road (I don’t know how to do it) is the notifications list should have a swipe to mark as read feature just like you can swipe discussion list items on mobile... on mobile it should be, swipe to mark as read, tap to go to the discussion.

**Review by askvortsov1 — APPROVED** [review]

File: less/forum/NotificationList.less [hunk 1]
```diff
@@ -103,7 +103,6 @@
     font-weight: bold;
     text-transform: uppercase;
   }
-
```
Comment (SychO9) [inline less/forum/NotificationList.less:106]: you removed a line by mistake

**Review by SychO9 — CHANGES_REQUESTED** [review]

**Comment by zerosonesfun** [comment]
Easier to re-do.

**Pull request closed without merging**